ansible_rulebook/auth.py

```diff
@@ -0,0 +1,51 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+import ssl
+import typing as tp
+
+import aiohttp
+
+from ansible_rulebook.conf import settings
+
+logger = logging.getLogger(__name__)
+
+
+async def renew_token() -> str:
+    logger.info(
+        "Renew websocket token from %s", settings.websocket_token_address
+    )
+    async with aiohttp.ClientSession() as session:
+        async with session.post(
+            settings.websocket_token_address,
+            data={"refresh": settings.websocket_refresh_token},
+            ssl_context=_sslcontext(),
+        ) as resp:
+            data = await resp.json()
+            if "access" not in data:
+                logger.error(f"Failed to renew token. Error: {str(data)}")
+                return ""
```

Comment:
@bzwei We should throw a well known exception here if we can't get the refresh token for any reason, e.g. the eda-server could be down for maintenance.